### PR TITLE
cannon: Implement bltzal

### LINKED ISSUE #12550: Implement bgezal and bltzal instructions
Implement the instructions that were previously sanitized [here](https://github.com/ethereum-optimism/optimism/blob/develop/cannon/Makefile#L45). Note that these instructions are displayed by objdump in their mnemonic form.

### PR DESCRIPTION
**Pull request by Inphi**
Implements the bltzal instruction. This is implemented for completeness even though the Go compiler doesn't seem to use this instruction.

**Note that this alters the STF of the 32-bit VMs. However, the Go compiler does not emit `bltzal` for 32-bit targets. So this does not alter the execution traces of 32-bit programs.**

## Metadata
Builds on https://github.com/ethereum-optimism/optimism/pull/12594
Fixes https://github.com/ethereum-optimism/optimism/issues/12550

### CODE REVIEW

**Comment by tynes** [comment]
@Inphi We haven't cut the release for holocene yet. Curious if you think we should cut the release before or after this is merged. I lean towards before

**Comment by Inphi** [comment]
> @Inphi We haven't cut the release for holocene yet. Curious if you think we should cut the release before or after this is merged. I lean towards before

My weak preference is that we cut the release after this is merged. Just so the onchain VM STF doesn't differ from what we have in the monorepo. But this shouldn't block the release if it's happening pretty soon.
fwiw, this will probably be the last single-threaded Cannon change before it's completely replaced by 64-bit/mt cannon.

**Comment by tynes** [comment]
We haven't cut the release yet, if you would like to merge this before then we just need to work out the security implications. We did a [small audit](https://gist.github.com/3docSec/068537844f2ddb204324079138f14551) for the previous change, we need to figure out if we need one for this too. I would think we do

**Comment by BlocksOnAChain** [comment]
@Inphi @tynes - if we decide to merge it before we cut a release for Holocene, I think we need @mds1 to share the security requirements, and we will potentially postpone some dates for Holocene. 
@mds1  - I would like to know what you think about this approach.
Also, if we add this change, are there some additional needs to audit the release with the Base team or to do an external audit?

**Review by mbaxter — APPROVED** [review]

**Comment by Inphi** [comment]
Also want to callout that a similar change that implements the `bgezal` instruction was recently merged via https://github.com/ethereum-optimism/optimism/pull/12569.

Whether or not this change is included on Holocene, it's not a blocker to merging this PR. If we decide not to include this for Holocene, we can cut a release on an older MIPS.sol contract that does not include the new changes. Let's continue the discussion on Holocene inclusion on slack/discord.

**Comment by tynes** [comment]
Looks like this needs a semver bump

**Pull request merged**
